include/display.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2017 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for display drivers and applications
+ */
+
+#ifndef __DISPLAY_H__
+#define __DISPLAY_H__
+
+/**
+ * @brief Display Interface
+ * @defgroup display_interface display Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <device.h>
+#include <stddef.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum display_pixel_format {
+	PIXEL_FORMAT_RGB_888 = 0,
+
+	PIXEL_FORMAT_END,
+};
+
+/**
+ * @typedef display_on_api
+ * @brief Callback API to turn display on
+ * See display_on() for argument description
+ */
+typedef int (*display_on_api)(struct device *dev);
+
+/**
+ * @typedef display_off_api
+ * @brief Callback API to turn display off
+ * See display_off() for argument description
+ */
+typedef int (*display_off_api)(struct device *dev);
+
+/**
+ * @typedef display_write_bitmap_api
+ * @brief Callback API for writing bitmap
+ * See display_write_bitmap() for argument description
+ */
+typedef int (*display_write_bitmap_api)(
```

Comment:
Moved from #9149
> @pizi-nordic
> Have you considered a bit more flexible approach based on bit blitting?

Reply:
No the intention of this particular api is only to write a flat bitmap, in case any bit blitting is needed it is up to the higher layer software to support this.

An additional api could be add in case HW support of some kind is available to perform the bit blitting.

Reply:
Do you think that a API for read operation might be useful?
Double buffering is the best option, however on memory constrained systems there might be not enough RAM. I think that such systems might be forced to sacrifice speed and read data from display in order to perform some operations.

There could be also systems, which have memory-mapped framebuffer. In such system reading is cheap and direct access is possible. I think that mmap()-like API (returning fixed pointer) might be the fastest solution in such case.

Reply:
Read API could be useful, I will add one but cant test it on the my current LCD as it does not allow read operations.

I could add a function to get a pointer to a memory-mapped framebuffer but is there any MCU that supports this?

Reply:
The EFM32 Giant Gecko Family has so called TFT Direct Drive:
>The EBI contains a TFT controller which can drive a TFT via a 565 RGB interface. The TFT controller supports programmable display
and port sizes and offers accurate control of frequency and setup and hold timing. Direct Drive is supported for TFT displays which do
not have their own frame buffer. In that case TFT Direct Drive can transfer data from either on-chip memory or from an external memo-
ry device to the TFT at low CPU load. Automatic alpha-blending and masking is also supported for transfers through the EBI interface

Details: https://www.silabs.com/documents/public/data-sheets/efm32gg-datasheet.pdf